File: dispersion-app/src/app/models/algorithms/random-with-leader-dispersion/RandomWithLeaderDispersionRobot.ts
import { Robot } from '../../core/Robot';
import { RobotState } from '../../utils/RobotState';

export class RandomWithLeaderDispersionRobot extends Robot {
  onID: number;
  destinationID: number;

  constructor(
    id: number,
    state: RobotState,
    onID: number,
    destinationID: number
  ) {
    super(id, state);
    this.onID = onID;
    this.destinationID = destinationID;
  }
}

File: java-engine/src/main/java/engine/algorithms/random_with_leader_dispersion/models/RandomWithLeaderDispersionRobot.java
package engine.algorithms.random_with_leader_dispersion.models;

import engine.core.models.Robot;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@EqualsAndHashCode(callSuper = false)
@AllArgsConstructor
@NoArgsConstructor
public class RandomWithLeaderDispersionRobot extends Robot {

    private Long onID;
    private Long destinationID;

}
